src/lsdb/core/crossmatch/kdtree_match.py

```diff
@@ -17,25 +19,36 @@ class KdTreeCrossmatch(AbstractCrossmatchAlgorithm):
     extra_columns = pd.DataFrame({"_dist_arcsec": pd.Series(dtype=np.dtype("float64"))})
 
     # pylint: disable=unused-argument,arguments-differ
-    def validate(self, n_neighbors: int = 1, radius_arcsec: float = 1, require_right_margin=True, **kwargs):
+    def validate(
+        self,
+        n_neighbors: int = 1,
+        radius_arcsec: float = 1,
+        min_radius_arcsec: float = 0,
+        require_right_margin=True,
+        **kwargs,
+    ):
         super().validate()
         validate_radius(radius_arcsec)
+        if min_radius_arcsec < 0:
+            validate_radius(min_radius_arcsec)
         if n_neighbors < 1:
             raise ValueError("n_neighbors must be greater than 1")
-
         # Check that the margin exists and has a compatible radius.
         if self.right_margin_hc_structure is None:
             if require_right_margin:
                 raise ValueError("Right margin is required for cross-match")
         else:
             if self.right_margin_hc_structure.catalog_info.margin_threshold < radius_arcsec:
                 raise ValueError("Cross match radius is greater than margin threshold")
+            if self.right_margin_hc_structure.catalog_info.margin_threshold < min_radius_arcsec:
+                raise ValueError("Cross match minimum radius is greater than margin threshold")
 
     # pylint: disable=unused-argument
     def crossmatch(
         self,
         n_neighbors: int = 1,
         radius_arcsec: float = 1,
+        min_radius_arcsec: float = 0,
```

Comment:
Can we have a unit test for a self match to make sure we're not ignoring matches with 0 distance between themself?